src/controllers/user.js

```diff
@@ -68,22 +68,26 @@ export const getAll = async (req, res) => {
 }
 
 export const updateById = async (req, res) => {
-  const paramsId = Number(req.params.id)
-  const { id } = req.user
-  const foundUser = await User.findById(paramsId)
+  const { firstName, lastName } = req.body
+  try {
+    validation.update(firstName, lastName)
+    const paramsId = Number(req.params.id)
+    const { id } = req.user
+    const canPatch = validation.validateCanModify(req)
+    if (!canPatch) {
+      return sendDataResponse(res, 403, { error: ERR.NOT_AUTHORISED })
+    }
+    const foundUser = await User.findById(paramsId)
+    if (!foundUser) {
+      return sendDataResponse(res, 404, { error: ERR.USER_NOT_FOUND })
+    }
+    const updatedUser = await User.updateUser(id, req.body)
```

Comment:
You're trying to see if a user exists based on the ID in the params, and then updating a user by the ID attached to the request. It's a little redundant.

Just because the route has an :id param doesn't mean you have to use it. I'd remove the foundUser stuff since the auth middleware will ensure that the user exists already, then just update the user based on the user ID attached to the request object